### PR TITLE
[0.6.0] Set up tests and fix read-only array length

### PR DESCRIPTION
**Pull request by erak**
Needed for https://github.com/ethereum/solidity/pull/7350.

This PR updates the workaround for breaking releases used by `npm test`. It also updates the newly added 0.6.0 tests to pass with `array.length` being an rvalue now.

### CODE REVIEW

File: test/DAO/DAO.sol [hunk 1]
```diff
@@ -447,8 +447,8 @@ contract DAO is DAOInterface, Token, TokenCreation {
         if (proposals.length == 1) // initial length is 1 (see constructor)
             lastTimeMinQuorumMet = now;
 
-        _proposalID = proposals.length++;
-        Proposal storage p = proposals[_proposalID];
+        Proposal storage p = proposals.push();
+        _proposalID = proposals.length;
```
Comment (leonardoalt) [inline test/DAO/DAO.sol]: Since this is done after the `push` I think it should be `length - 1`

**Comment by leonardoalt** [comment]
@erak shouldn't the target be `master_060`?

**Comment by erak** [comment]
After the conversion on Friday, I thought we could try to get away without having a separate 0.6.0 branch. Would it be OK to have the new 0.6.0 determinism contracts in a 0.5.12 release of solc-js?

**Comment by erak** [comment]
@leonardoalt https://github.com/ethereum/solidity/pull/7462 needs to be adjusted depending on the above as well.

**Comment by leonardoalt** [comment]
> Would it be OK to have the new 0.6.0 determinism contracts in a 0.5.12 release of solc-js?

I think this would be a bit weird, wouldn't it? To me having a separate `master_060` feels more organized.
@axic ?

**Comment by leonardoalt** [comment]
I'll merge for now, but let's write a PR next to extract the .sol tests to the Solidity repo.

**Pull request merged**